### PR TITLE
The --ports flag does not modify dc env variables

### PR DESCRIPTION
**Pull request by pecameron**
--ports is not intended to modify the env variables. That can
be done using oc env or by editing the router's dc. Rather it
sets the ports that are exposed in the router's service.

The help message is confusing and misleading.  This change
rewords the help text.

bug 1420543
https://bugzilla.redhat.com/show_bug.cgi?id=1420543

### CODE REVIEW

**Comment by pecameron** [comment]
@knobunc PTAL

**Comment by pecameron** [comment]
@rajatchopra PTAL

**Comment by pecameron** [comment]
@rajatchopra @knobunc PTAL

**Comment by rajatchopra** [comment]
How is this related to host-network=true/false?

File: pkg/cmd/admin/router/router.go [hunk 1]
```diff
@@ -276,7 +276,7 @@ func NewCmdRouter(f *clientcmd.Factory, parentName, name string, out, errout io.
 	cmd.Flags().StringVar(&cfg.ForceSubdomain, "force-subdomain", "", "A router path format to force on all routes used by this router (will ignore the route host value)")
 	cmd.Flags().StringVar(&cfg.ImageTemplate.Format, "images", cfg.ImageTemplate.Format, "The image to base this router on - ${component} will be replaced with --type")
 	cmd.Flags().BoolVar(&cfg.ImageTemplate.Latest, "latest-images", cfg.ImageTemplate.Latest, "If true, attempt to use the latest images for the router instead of the latest release.")
-	cmd.Flags().StringVar(&cfg.Ports, "ports", cfg.Ports, "A comma delimited list of ports or port pairs to expose on the router pod. The default is set for HAProxy. Port pairs are applied to the service and to host ports (if specified).")
```
Comment (knobunc) [inline pkg/cmd/admin/router/router.go]: Service isn't right is it?  It controls what goes into the pod spec.

**Comment by knobunc** [comment]
@rajatchopra it only matters when hostNetwork is false because when it is true, the port mapping in

**Comment by knobunc** [comment]
Bah.  Wrong button.  @rajatchopra it only matters when hostNetwork is false because this controls the port mappings in the pod spec that docker sets up.  And those only matter when hostNetwork is true.  At least, that's what I think is the case.  @pecameron can you post the difference between things when --ports is set vs. when it is not?

File: pkg/cmd/admin/router/router.go (hunk 1 above)
Comment (pecameron) [inline pkg/cmd/admin/router/router.go]: It appears in the router pod containerPort and hostPort:
    ports:
    - containerPort: 1080
      hostPort: 1080
      protocol: TCP
It also appears in the service:
exampleport         172.30.138.192   <none>        1080/TCP,1935/TCP,1936/TCP               4m
How do you want to phrase this?

**Comment by pecameron** [comment]
@knobunc --ports sets the port in the router pod containerPort and hostPort. It also sets service port and targetPort.

**Comment by pecameron** [comment]
@knobunc What are we going to do with this?

**Comment by pecameron** [comment]
@knobunc PTAL

**Comment by pecameron** [comment]
@knobunc Is this ready to go?

File: pkg/cmd/admin/router/router.go (hunk 1 above)
Comment (knobunc) [inline pkg/cmd/admin/router/router.go]: "A comma delimited list of ports or port pairs that set the port in the router pod containerPort and hostPort. These are also used as the service port and targetPort when the router service is created. Note: this does not modify what the router will actually listen on. That can be done by editing the router environment variables. This is meaningful when host-network=false."

**Comment by knobunc** [comment]
@openshift/networking FYI

**Comment by knobunc** [comment]
Will merge on Monday.

**Comment by knobunc** [comment]
[merge]

**Comment by knobunc** [comment]
[merge]

**Comment by knobunc** [comment]
[mrege] last flaked on https://github.com/openshift/origin/issues/14043 (logs: https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_origin/1153/)

**Pull request merged**